src/InstallDir/installDirReducer.ts

```diff
@@ -46,65 +46,52 @@ import { RootState } from '../state';
 const CONFIRM_DIR = Symbol('Confirm the install directory');
 const SET_DIR = Symbol('Set the install directory');
 
-const SET_INSTALL_DIR = 'SET_INSTALL_DIR';
-export const setInstallDir = (dir: string) => ({
-    type: SET_INSTALL_DIR,
-    dir,
-});
-
-const SHOW_CONFIRM_INSTALL_DIR_DIALOG = 'SHOW_CONFIRM_INSTALL_DIR_DIALOG';
-export const showConfirmInstallDirDialog = (versionToInstall: string) => ({
-    type: SHOW_CONFIRM_INSTALL_DIR_DIALOG,
-    versionToInstall,
-});
-
-const SHOW_SET_INSTALL_DIR_DIALOG = 'SHOW_SET_INSTALL_DIR_DIALOG';
-export const showSetInstallDirDialog = () => ({
-    type: SHOW_SET_INSTALL_DIR_DIALOG,
-});
-
-const HIDE_INSTALL_DIR_DIALOG = 'HIDE_INSTALL_DIR_DIALOG';
-export const hideInstallDirDialog = () => ({
-    type: HIDE_INSTALL_DIR_DIALOG,
-});
-
-const initialState = () => ({
+type InstallDirectoryState = {
+    currentDir: string;
+    isDialogVisible: boolean;
+    dialogFlavour?: symbol;
+    versionToInstall?: string;
+};
+const initialState: InstallDirectoryState = {
     currentDir: persistedInstallDir(),
```

Comment:
Please check, whether the persisted install dir is really used here correctly. I suspect it is not, because … of complicated reasons. ;-) 

I can explain it in person why that happens, but currently something like a top-level 
`const x = persistedInstallDir()` usually does not work correctly, while a function like `const x = () => persistedInstallDir()` usually does work correctly.

Reply:
I couldn't get this to fail, it worked nicely when testing again now. Looking forward to learn some quirky bits on why it needs to be invoked :)